react-task/src/components/Input.jsx

```diff
@@ -1,29 +1,57 @@
 import React from "react";
 import PropTypes from "prop-types";
 
-const Input = ({ type, id, placeholder, helpText }) => (
-  <div class="form-floating mb-3">
-    <input
-      type={type}
-      class="form-control"
-      id={id}
-      placeholder={placeholder}
-      aria-describedby={helpText && "help"}
-    ></input>
-    <label for={id}>{placeholder}</label>
-    {helpText && (
-      <div id="help" class="form-text">
-        {helpText}
-      </div>
-    )}
-  </div>
-);
+const Input = ({
+  type,
+  id,
+  placeholder,
+  helpText,
+  value,
+  handleDataChange,
+  warningText,
+}) => {
+  const handleOnChange = (event) => {
+    const target = event.target.name;
+    const value = event.target.value;
+    handleDataChange((prevValue) => ({ ...prevValue, [target]: value }));
+  };
+
+  return (
+    <div class="form-floating mb-3">
+      <input
+        type={type}
+        class="form-control"
+        id={id}
+        placeholder={placeholder}
+        value={value}
+        name={id}
+        onChange={(event) => handleOnChange(event)}
```

Comment:
You can update this as follows. Adding the enclosing function is redundant and we can just use the function reference

```suggestion
        onChange={handleOnChange}
```